waltz-tools/src/main/java/com/wepay/waltz/tools/cluster/ClusterCli.java

```diff
@@ -289,20 +301,170 @@ private void failMissingValidationResults(List<PartitionValidationResults> parti
+            }
+        }
+
+        private StorageAdminClient openStorageAdminClient(String storageHost,
+                                                         Integer storageAdminPort,
+                                                         SslContext sslContext,
+                                                         StoreMetadata storeMetadata) throws Exception {
+            StoreParams storeParams = storeMetadata.getStoreParams();
+            StorageAdminClient storageAdminClient = null;
+            try {
+                 storageAdminClient = new StorageAdminClient(storageHost, storageAdminPort, sslContext,
+                    storeParams.key, storeParams.numPartitions);
+                storageAdminClient.open();
+
+            } catch (Exception e) {
+                // Do Nothing.
+            }
+            return storageAdminClient;
+        }
+
+        private Map<Integer, Map<String, Boolean>> getPartitionStatusFromStorage(StoreMetadata storeMetadata,
+                                                   Map<String, Integer> storageConnections,
+                                                   SslContext sslContext,
+                                                   int numPartitions) throws Exception {
+            Map<Integer, Map<String, Boolean>> partitionToStorageStatusMap = new HashMap<>();
+            List<StorageAdminClient> storageAdminClients = new ArrayList<>();
+            List<CompletableFuture<Object>> futures = new ArrayList<>();
+
+            for (int id = 0; id < numPartitions; id++) {
+                partitionToStorageStatusMap.put(id, new HashMap<>());
+            }
+
+            for (Map.Entry<String, Integer> storageNodeConnection : storageConnections.entrySet()) {
+                String[] storageHostAndPortArray = storageNodeConnection.getKey().split(":");
+                String storageHost = storageHostAndPortArray[0];
+
+                Integer storageAdminPort = storageNodeConnection.getValue();
+
+                StorageAdminClient storageAdminClient = openStorageAdminClient(storageHost, storageAdminPort,
+                    sslContext, storeMetadata);
+
+                if (storageAdminClient.isValid()) {
+                    CompletableFuture<Object> future = storageAdminClient.getAssignedPartitionStatus()
+                        .handle((obj, th) -> {
+                            if (th == null) {
+                                Map<Integer, Boolean> partitionStatusMap = (Map<Integer, Boolean>) obj;
+
+                                synchronized (partitionToStorageStatusMap) {
+                                    partitionStatusMap.forEach((partitionId, status) -> {
+                                        partitionToStorageStatusMap.get(partitionId).put(storageNodeConnection.getKey(), status);
+                                    });
+                                }
+                            }
+                            return null;
```

Comment:
You may use `whenComplete` as this callback is not generating any "new" type of return value.